### PR TITLE
Add more rules to CrossCompile

### LINKED ISSUE #75: Iterator#sameElements has a different signature between 2.12 and 2.13
`iterable.sameElements(iterable)` compiles in 2.12, but in 2.13 the recommended way to write this is to use `iterable.iterator.sameElements(iterable)`. However, that form does not cross-compile with 2.12, because `sameElements` take an `Iterator` in 2.12, whereas it takes an `IterableOnce` in 2.13. We can produce 2.12-compatible code by producing `iterable.iterator.sameElements(iterable.iterator)` instead.

### PR DESCRIPTION
**Pull request by MasseGuillaume**
fix #75

### CODE REVIEW

**Comment by MasseGuillaume** [comment]
@julienrf ready for review

File: compat/src/main/scala-2.11_2.12/scala/collection/compat/package.scala [hunk 1]
```diff
@@ -54,6 +54,18 @@ package object compat {
...
+  implicit class IteratorExtensionMethods[A](private val self: Iterator[A]) extends AnyVal {
+    def sameElements[B >: A](that: IterableOnce[B]): Boolean = {
+      val those = that.iterator
+      while (self.hasNext && those.hasNext)
+        if (self.next() != those.next())
+          return false
+      // At that point we know that *at least one* iterator has no next element
+      // If *both* of them have no elements then the collections are the same
+      self.hasNext == those.hasNext
+    }
```
Comment (julienrf) [inline compat/src/main/scala-2.11_2.12/scala/collection/compat/package.scala]: You can simply delegate to `self.sameElements(that.iterator)`.

Comment (julienrf) [reply compat/src/main/scala-2.11_2.12/scala/collection/compat/package.scala]: Can you add a test for this decorator in `compat/src/test/scala/`?

File: scalafix/rules/src/main/scala/fix/CanBuildFrom.scala [hunk 2]
```diff
@@ -4,6 +4,8 @@ import scalafix._
 import scalafix.util._
 import scala.meta._
 
+import scala.collection.mutable.{Set => MSet}
```
Comment (julienrf) [inline scalafix/rules/src/main/scala/fix/CanBuildFrom.scala]: The recommended practice is to import `scala.collection.mutable` and then write `mutable.Set` at use site.

File: scalafix/rules/src/main/scala/fix/CanBuildFrom.scala [hunk 3]
```diff
@@ -153,6 +157,8 @@ case class CanBuildFromNothing(param: Name,
       body.collect {
         case ap @ Term.ApplyType(Term.Select(e, to @ toTpe(_)), List(cc2 @ matchCC(_))) =>
 
+          handledTo += to
```
Comment (julienrf) [inline scalafix/rules/src/main/scala/fix/CanBuildFrom.scala]: How is this code related to the `.sameElements` rewrite rule?

Comment (julienrf) [reply scalafix/rules/src/main/scala/fix/CanBuildFrom.scala]: Can you explain the list of changes performed by this commit in the commit message?

**Comment by MasseGuillaume** [comment]
This one is ready for merging.

**Comment by julienrf** [comment]
Do you mind adding more information about the changes in the commit message?

**Comment by MasseGuillaume** [comment]
Done
